scripts/ci.sh

```diff
@@ -17,9 +17,12 @@
 set -e
 set -x
 
+# Set go-langs data race detector options
+export GORACE="halt_on_error=1"
```

Comment:
would be good to know the reason for this either as a comment or in the commit msg, because it is not the default and it seems normal to continue running other unit tests to gather all problems.

Reply:
I thought it would be reasonable to stop as soon as this fails but if you think it's better to just run all tests even if there is a race detection fail let me know I can remove it. Otherwise I'm not sure what to comment the `halt_on_error` is pretty self documenting right ?

Reply:
I think if this fails its hard to see other failures scrolling through the logs in GH. Or you just see that test `xyz` failed, then fix it and push again, just to learn that you did not scroll up enough to see that a race detection failed too, so you'll have to go back and fix/push again.